public/app/plugins/datasource/loki/components/monaco-query-field/MonacoQueryField.tsx

```diff
@@ -138,6 +149,24 @@ const MonacoQueryField = ({ history, onBlur, onRunQuery, initialValue, datasourc
     editor.onDidChangeModelContent(checkDecorators);
   };
 
+  let typingTimer: NodeJS.Timeout;
+
+  const onType = (query: string) => {
```

Comment:
Instead of creating a timer, you could use debounce https://lodash.com/docs/4.17.15#debounce which `Creates a debounced function that delays invoking func until after wait milliseconds have elapsed since the last time the debounced function was invoked`

Reply:
Btw, same can be done with RxJS: https://rxjs.dev/api/operators/debounce